spec/connectors/registry_spec.rb

```diff
@@ -9,21 +9,79 @@
 require 'connectors/registry'
 
 describe Connectors::Factory do
-  let(:factory) do
-    described_class.new
+  subject { described_class.new }
+
+  let(:configuration) {
+    {}
+  }
+
+  let(:job_description) {
+    {}
+  }
+
+  let(:registered_connector) {
+    'my-connector'
+  }
+
+  let(:unregistered_connector) {
+    'another-connector'
+  }
+
+  class MyConnector
+    def initialize(configuration: {}, job_description: {})
+      # nothing needed
+    end
+  end
+
+  before(:each) do
+    subject.register(registered_connector, MyConnector)
   end
 
   describe '#connector_class' do
-    it 'let us register a connector under a name' do
-      class MyConnector
-        def works
-          'works'
-        end
+    it 'returns registered class' do
+      expect(subject.connector_class(registered_connector)).to eq MyConnector
+    end
+  end
+
+  describe '#registered?' do
+    context 'my-connector is registered' do
+      it 'should return that my-connector is registered' do
+        expect(subject.registered?(registered_connector)).to be_truthy
+      end
+
+      it 'should return that another-connector is not registered' do
+        expect(subject.registered?(unregistered_connector)).to be_falsey
       end
+    end
+  end
+
+  describe '#connector' do
+    context 'my-connector is registered' do
+      it 'should return a my-connector instance' do
+        connector_instance = subject.connector(registered_connector, configuration, job_description)
+
+        expect(connector_instance).to be_a(MyConnector)
+      end
+    end
+
+    context 'another-connector is not registered' do
+      it 'should raise an exception, that another-connector is not registered' do
+        expect { subject.connector(unregistered_connector, configuration, job_description) }.to raise_exception
+      end
+    end
+  end
+
+  describe '#registered_connectors' do
+    let(:registered_connectors) {
+      %w[a-connector b-connector c-connector]
+    }
+
+    before(:each) do
+      registered_connectors.each { |connector| subject.register(connector, MyConnector) }
+    end
 
-      factory.register('sharepoint', MyConnector)
-      connector_class = factory.connector_class('sharepoint')
-      expect(connector_class.new.works).to eq 'works'
+    it 'returns registered connectors' do
+      expect(subject.registered_connectors).to include('a-connector', 'b-connector', 'c-connector')
```

Comment:
```suggestion
      expect(subject.registered_connectors).to include(*registered_connectors)
```

I think array splat operator usage here is a good way to make test simpler and robust :)